client/influxdb.go

```diff
@@ -195,9 +196,13 @@ func (c *Client) Query(q Query) (*Response, error) {
 // QueryContext sends a command to the server and returns the Response
 // It uses a context that can be cancelled by the command line client
 func (c *Client) QueryContext(ctx context.Context, q Query) (*Response, error) {
-	u := c.url
+	// clone url struct
```

Comment:
I don't like this cloning the struct thing. The original line, `u := c.url`, already does a soft clone of the struct without the additional requirement to render a string and then reparse that string back into the struct. The only thing that doesn't get cloned properly is the `.User` variable on the struct, but that isn't being modified so a shallow clone should be fine.

It seems to me like it would be better to just modify line 200 in the original diff to:

```
u.Path += "/query"
```

And that's all that would be required to implement this feature. If the path is empty, the path should still be preceded by a slash anyway and it outputs the correct output. See this example: https://play.golang.org/p/veRZfeI9Grj

Reply:
soft clone did not work, it is a pointer:
https://play.golang.org/p/X9vztSPdo4P

Reply:
???

Reply:
Sorry, I didn't see you had replied.

If you look at the struct definition, you will see this:

```go
type Client struct {
    url        url.URL
    unixSocket string
    username   string
    password   string
    httpClient *http.Client
    userAgent  string
    precision  string
}
```

If you notice, `url.URL` is not a pointer. So when you do this:

```
u := c.url
```

That creates a copy of the `url.URL` struct. We can then look at [url.URL](https://godoc.org/net/url#URL) in the documentation and we see that almost all of the attributes in there are values. There's one pointer, but we don't modify it so it doesn't matter. That means it is safe for us to modify without causing any kind of race condition.

So my original statement stands. It would be easier to modify the original code to just add the additional path component rather than to do a complex rendering of the url to a string and then reparsing it while checking for an error that would never happen.

Reply:
oh sry, u are right

Reply:
https://play.golang.org/p/8FHZDXpLnK_c